### PR TITLE
update namespace selector for topology webhook

### PR DESCRIPTION
**Pull request by kon-angelo**
**How to categorize this PR?**

/area control-plane
/kind bug
/platform azure

**What this PR does / why we need it**:

After https://github.com/gardener/gardener/pull/6997, handling only the shoot namespaces in the topology webhook is not enough. We should extend the applicable namespaces while excluding critical ones to prevent deadlocks.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```breaking operator
Update the azure topology webhook to watch all namespaces and not just shoot namespaces. The `kube-system`, `garden` and extension namespaces are except to prevent deadlocks.
```

### CODE REVIEW

**Review by MartinWeindel — APPROVED** [review]
/lgtm

File: pkg/webhook/topology/add.go [hunk 1]
```diff
@@ -51,9 +51,17 @@ func AddToManager(mgr manager.Manager) (*extensionswebhook.Webhook, error) {
...
-				v1beta1constants.LabelSeedProvider: azure.Type,
-				v1beta1constants.GardenRole:        v1beta1constants.GardenRoleShoot,
+			MatchExpressions: []metav1.LabelSelectorRequirement{
+				{
+					Key:      v1beta1constants.GardenRole,
+					Operator: metav1.LabelSelectorOpNotIn,
+					Values:   []string{v1beta1constants.GardenRoleExtension},
+				},
+				{
+					Key:      "kubernetes.io/metadata.name",
```
Comment (timuthy) [inline pkg/webhook/topology/add.go:61]:
```suggestion
					Key:      corev1.LabelMetadataName
```

**Review by timuthy — COMMENTED** [review]
One nit. Can you please add a proper release note?

File: pkg/webhook/topology/add.go (hunk 1 above)
Comment (kon-angelo) [reply pkg/webhook/topology/add.go:61]: Ah, I tried to find the const for this but I failed. Thanks for the catch

**Comment by timuthy** [comment]
Can you please check the failing `verify` step @kon-angelo?

**Review by timuthy — APPROVED** [review]
/lgtm
Thanks for taking care!

**Pull request merged**